pkg/aws_cloudwatch.go

```diff
@@ -360,39 +360,32 @@ func createPrometheusLabels(cwd *cloudwatchData, labelsSnakeCase bool) map[strin
 	return labels
 }
 
-func ensureLabelConsistencyForMetrics(metrics []*PrometheusMetric) []*PrometheusMetric {
-	labelSetForMetric := make(map[string]map[string]bool, len(metrics))
```

Comment:
This is also another source of unneeded allocations caused by me when I refactored this. The `len(metrics)` is likely to be much much larger than the collection of unique metric names. The larger the scrape the more over-allocated this would have been.